pkg/cluster/k8sres.go

```diff
@@ -2195,8 +2207,22 @@ func (c *Cluster) generateLogicalBackupPodEnvVars() []v1.EnvVar {
 
 	if c.OpConfig.LogicalBackup.LogicalBackupS3SecretAccessKey != "" {
 		envVars = append(envVars, v1.EnvVar{Name: "AWS_SECRET_ACCESS_KEY", Value: c.OpConfig.LogicalBackup.LogicalBackupS3SecretAccessKey})
+	} else if c.OpConfig.LogicalBackup.LogicalBackupS3AccessSecretName != "" {
```

Comment:
```suggestion
	}
	if c.OpConfig.LogicalBackup.LogicalBackupS3AccessSecretName != "" {
```

We want `LogicalBackupS3AccessSecretName` to take precedence over `LogicalBackupS3SecretAccessKey`. So swap the if blocks and in for `LogicalBackupS3SecretAccessKey` use the `appendEnvVars` function, which only appends if `envVars` does not contain `AWS_SECRET_ACCESS_KEY`